web/src/components/AddressAutocomplete/AddressAutocomplete.js

```diff
@@ -0,0 +1,160 @@
+import React, { useState, useEffect, useRef } from 'react'
+import { TextField, Label } from '@redwoodjs/forms'
+
+const AddressAutocomplete = (props) => {
+  const address1FieldRef = useRef(null)
+  const [addressStreet, setAddressStreet] = useState(props.addressStreet || '')
+  const [postcode, setPostcode] = useState('')
+  const [statelocation, setStatelocation] = useState('')
+  const [localCity, setLocalCity] = useState('')
+  const [country, setCountry] = useState('')
+  const [addressUnitNumber] = useState('')
+
+  useEffect(() => {
+    const handleScriptLoad = (address1FieldRef) => {
+      const autoComplete = new window.google.maps.places.Autocomplete(
+        address1FieldRef.current,
+        { fields: ['address_components', 'geometry'], types: ['address'] }
+      )
+
+      autoComplete.setFields(['address_components', 'formatted_address'])
+      autoComplete.addListener('place_changed', () =>
+        handlePlaceSelect(autoComplete)
+      )
+    }
+    loadScript(
+      `https://maps.googleapis.com/maps/api/js?key=${process.env.GOOGLE_MAP_API_KEY}&libraries=places`,
+      () => handleScriptLoad(address1FieldRef)
+    )
+  }, [])
+
+  const loadScript = (url, callback) => {
+    let script = document.createElement('script')
+    script.type = 'text/javascript'
+
+    if (script.readyState) {
+      script.onreadystatechange = function () {
+        if (
+          script.readyState === 'loaded' ||
+          script.readyState === 'complete'
+        ) {
+          script.onreadystatechange = null
+          callback()
+        }
+      }
+    } else {
+      script.onload = () => callback()
+    }
+
+    script.src = url
+    document.getElementsByTagName('head')[0].appendChild(script)
+  }
+
+  const handlePlaceSelect = async (autoComplete) => {
+    const addressObject = autoComplete.getPlace()
+    let streetAddress = ''
+    let postcode = ''
+    for (const component of addressObject.address_components) {
+      const componentType = component.types[0]
+
+      switch (componentType) {
+        case 'street_number': {
+          streetAddress += component.long_name
+          break
+        }
+
+        case 'route': {
+          streetAddress = streetAddress + ' ' + component.short_name
```

Comment:
Can we use string interpolation here?

```suggestion
          streetAddress = `${streetAddress} ${component.short_name}`
```